elastic/kibana
# [Upgrade Assistant] Hide system indices from es deprecations list

Fixes https://github.com/elastic/kibana/issues/113280

### Summary
Currently UA shows in the ES deprecations page deprecations for system indices, and since for 7.16 system indices will be resolved separately through the system indices migration flow that will be shown in the Overview page (see https://github.com/elastic/kibana/pull/110593) we have to remove them from the es deprecations table.

**How to test**

1. Unzip [6.8-data-snapshot.zip](https://github.com/elastic/kibana/files/7242690/6.8-data-snapshot.zip) and start up ES as follows: `yarn es snapshot -E path.data=/path/to/6.8-data-snapshot`. This contains indices and ML job model snapshots created on 6.8. This will trigger the reindex and ML deprecations.
2. Since ES system indices upgrade API returns a mock for now, change [this line](https://github.com/sabarasaba/kibana/blob/ua/hide_system_indices_es_deprecations_table/x-pack/plugins/upgrade_assistant/server/lib/es_deprecations_status.ts#L30) to: 
```
    const systemIndicesList = [
      ...convertFeaturesToIndicesArray(systemIndices.features),
      '.ml-config',
    ];
```
3. Start up Kibana, `yarn start`.
4. Navigate to UA, open `ES deprecation issues`
5. Verify that there is no deprecation that needs reindex for `.ml-config`
6. Verify that the stats widget in the overview page also reflect the missing index in the count

If after following the steps you don't see any deprecations for es in upgrade assistant, try unzipping the folder in the `kibana/.es` folder and starting up es with: `yarn es snapshot -E path.data=../6.8-data-snapshot`
